src/Admin/Model/DoctrineMetadataServiceResource.php

```diff
@@ -1,57 +1,64 @@
 <?php
 /**
  * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
- * @copyright Copyright (c) 2013 Zend Technologies USA Inc. (http://www.zend.com)
+ * @copyright Copyright (c) 2013-2016 Zend Technologies USA Inc. (http://www.zend.com)
  */
 
 namespace ZF\Apigility\Doctrine\Admin\Model;
 
+use Doctrine\Common\Persistence\Mapping\AbstractClassMetadataFactory;
+use Interop\Container\ContainerInterface;
+use ZF\Apigility\Admin\Model\RestServiceEntity;
 use ZF\ApiProblem\ApiProblem;
 use ZF\Rest\AbstractResourceListener;
-use Zend\ServiceManager\ServiceManager;
-use Exception;
 
 class DoctrineMetadataServiceResource extends AbstractResourceListener
 {
+    /**
+     * @var ContainerInterface
+     */
     protected $serviceManager;
 
-    public function setServiceManager(ServiceManager $serviceManager)
+    /**
+     * @param ContainerInterface $serviceManager
+     * @return $this
+     */
+    public function setServiceManager(ContainerInterface $serviceManager)
```

Comment:
Actually, the original is incorrect, too. :frowning: (should have been `setServiceLocator(ServiceLocatorInterface $serviceLocator)`).

Regardless, though, changing the typehint represents a BC break, so revert to the original.